### ISSUE TITLE
tensorflow.experimental.numpy.lcm giving inconsistent results

### ISSUE THREAD

**Issue by hirwa-nshuti** [tensorflow/tensorflow#58955]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

tf 2.9.1

### Custom Code

Yes

### OS Platform and Distribution

Linux ubuntu 18.04

### Python version

3.8

### Current Behaviour?

```shell
I am trying to use the function `tf.experimental.numpy.lcm` but getting unusual results after doing a manual calculation.

Here is an example:
x = tf.constant([6], dtype=tf.int8)
y = tf.constant([22], dtype=tf.int8)

The expected lcm of the above two tensors should be a tensor `tf.Tensor([66], shape=(1,), dtype=int8)`
```

### Standalone code to reproduce the issue

Using Tensorflow

```

import tensorflow as tf

x = tf.constant([6], dtype=tf.int8)
y = tf.constant([22], dtype=tf.int8)
print(tf.experimental.numpy.lcm(x, y)) # Current output: tf.Tensor([62], shape=(1,), dtype=int8)
```
Using numpy

```
import numpy as np

x = np.array([6], dtype=np.int8)
y = np.array([22], dtype=np.int8)
z = np.lcm(x, y)
print(repr(z)) # Output: array([66], dtype=int8)
```
```


### Relevant log output

_No response_</details>

**Comment by tiruk007** [comment]
@SuryanarayanaY 
I was able to reproduce the issue on Colab using TF v2.11. Please find the gist [here](https://colab.research.google.com/gist/tiruk007/e66b3642663832cd2dfd5e88397f4a30/untitled47.ipynb) for reference.

Thank you!

**Comment by SuryanarayanaY** [comment]
Hi @hirwa-nshuti ,
Thanks for reporting. We have observed the behaviour of this API recently in #58619 and reported the same to Developer Team for action.

Thankyou!

**Comment by SuryanarayanaY** [comment]
Hi @hirwa-nshuti ,

The issue has been fixed in tf-nightly.Please refer to the attached [gist](https://colab.research.google.com/gist/SuryanarayanaY/4b7e2524c1945dcbaa22d3fe5c14299a/58955-resolved.ipynb).

Could you please crosscheck and confirm from your side? Please feel free to close if resolved. Thanks!

**Comment by hirwa-nshuti** [comment]
I am running the gist from my end but getting a runtime error🧐
```
RuntimeError                              Traceback (most recent call last)
RuntimeError: module compiled against API version 0xf but this version of numpy is 0xe
```

**Comment by SuryanarayanaY** [comment]
@hirwa-nshuti ,

The error is related to numpy version mismatch in Colab environment which appears some times. Could you please try upgrading numpy version to 1.22.4 using pip and then restart runtime in colab and check whether numpy version upgraded to 1.22.4. Once done you can test the code. 

Please find the [gist](https://colab.research.google.com/gist/SuryanarayanaY/1f47aab2a8ec456c2733d3e2128cd08e/58955-resolved_r2.ipynb) where I followed the above steps and able to resolve the reported runtime error and also the main reported issue addressed with latest nightly_version(tf_nightly-2.13.0.dev20230220). 

Please check and close the issue if resolved. Thanks again for reporting the issue .

**Comment by hirwa-nshuti** [comment]
Hi @SuryanarayanaY 
Thanks for fixing the issue I tested the code from my end and it works as expected🙂
